### PR TITLE
Bug Fix: Sample Project reloads on empty form submission

### PR DESCRIPTION
**Pull request by sthodup1**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`grunt test`).
5. Make sure your code lints (`grunt lint`) - we've done our best to make sure these rules match our internal linting guidelines.
6. If you haven't already, complete the [CLA](https://code.facebook.com/cla).

Returning false ensures that the page does not reload when one of the input fields is empty. Consequently, this prevents form data from being cleared on page reload.

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: docs/docs/tutorial.md [hunk 1]
```diff
@@ -549,7 +549,7 @@ var CommentForm = React.createClass({
     var author = this.state.author.trim();
     var text = this.state.text.trim();
     if (!text || !author) {
-      return;
+      return false;
```
Comment (jimfb) [inline docs/docs/tutorial.md]: Does this actually change anything?  I thought we started ignoring the return value of an event handler a long time ago.

Comment (sthodup1) [inline docs/docs/tutorial.md]: @jimfb It made a difference when I ran it in my browser. Without the return false, the page reloaded and the form cleared.

**Comment by facebook-github-bot-5** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: docs/docs/tutorial.md (hunk 1 above)
Comment (Aweary) [inline docs/docs/tutorial.md]: [Here's a jsfiddle](https://jsfiddle.net/69z2wepo/46290/) with the current code, I'm not seeing any difference when the `return` value is changed. What version of React are you using?

Comment (sthodup1) [inline docs/docs/tutorial.md]: Was using 15.0.1 but switching to the newest version fixed the problem. The zip included with the tutorial uses 15.0.1, which may be the cause of the issue.

Comment (jimfb) [inline docs/docs/tutorial.md]: @sthodup1 Can you try with the latest version?  If you're able to reproduce, can you provide a jsfiddle which demonstrates the problem/fix?

Comment (Aweary) [inline docs/docs/tutorial.md]: Glad to here. Maybe you can open a PR that updates the example's React dependency?

Comment (sthodup1) [inline docs/docs/tutorial.md]: @jimfb @Aweary The newest version fixes the issue. I just submitted a PR updating the example's React dependency.

Comment (Aweary) [inline docs/docs/tutorial.md]: Thanks @sthodup1! would you mind closing this PR then? 👍

**Comment by jimfb** [comment]
As per discussion above, closing in favor of https://github.com/reactjs/react-tutorial/pull/144

**Pull request closed without merging**